BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java

```diff
@@ -84,13 +84,15 @@ public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
   @Override
   public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
     Long id = data.get(position).getId();
+    String text = noteModel.getNoteById(id).getText();
```

Comment:
- Why is the noteModel in the NoteRecyclerViewAdapter? The model and the view should be separated and controlled by the Fragment. The NoteRecyclerViewAdapter should know nothing about the model. It should use the noteList from the NotesFragment when created, have a look at the other adapters. 

- The rest of the code looks fine to me.

Reply:
This was a workaround since the adapter needs to fetch and provide the note text to the TextEditor, but NoteItems did not contain a text yet and name is not identical, but only it's first line. 
This is causing trouble as well since the adapter needs to delete notes as well as the noteItems on swipe left or on select and using the delete-panel.

The noteModel-usages have been removed from the adapter though - I adjusted the NoteItems to contain a text parameter. The actual deletion of notes as view and as data from the database is done through the NotesFragment with a noteModel.